### PR TITLE
add doctests for example 2/n segmentation

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Adding doctest for `SegModel`, follow-up #5079 

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pl_examples/domain_templates/semantic_segmentation.py [hunk 1]
```diff
@@ -17,6 +17,29 @@
 DEFAULT_VALID_LABELS = (7, 8, 11, 12, 13, 17, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 31, 32, 33)
 
 
+def _create_synth_Cityscapes_dataset(path_dir):
```
Comment (justusschock) [inline pl_examples/domain_templates/semantic_segmentation.py:20]: Why do we need this? This isn't used?

Comment (Borda) [reply pl_examples/domain_templates/semantic_segmentation.py:20]: it is til wip, and the point here is to create a fake dataset if two images and skip downloading full dataset...

File: pl_examples/domain_templates/semantic_segmentation.py [hunk 2]
```diff
@@ -17,6 +17,29 @@
...
+
+    for split in dataset_splits:
+        for city in non_existing_citites:
+            (images_dir / split / city).mkdir(parents=True)
+            (fine_labels_dir / split / city).mkdir(parents=True)
+            base_name = str(uuid.uuid4())
+            image_name = f'{base_name}_leftImg8bit.png'
+            instance_target_name = f'{base_name}_gtFine_instanceIds.png'
+            semantic_target_name = f'{base_name}_gtFine_labelIds.png'
+            Image.new('RGB', (2048, 1024)).save(
```
Comment (tchaton) [inline pl_examples/domain_templates/semantic_segmentation.py:35]: Do you need the image to be 2048 * 1024 ?

Comment (Borda) [reply pl_examples/domain_templates/semantic_segmentation.py:35]: pls ignore this func, it is till very WIP :D

**Review by tchaton — APPROVED** [review]

**Comment by justusschock** [comment]
Sorry @tchaton clicked the wrong button

**Review by justusschock — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Pull request merged**